src/RevTracer.cc

```diff
@@ -182,6 +182,25 @@ void RevTracer::memRead(uint64_t adr, size_t len, void *data)
     traceRecs.emplace_back(TraceRec_t(MemLoad,adr,len,d));
 }
 
+void SST::RevCPU::RevTracer::memhSendRead(uint64_t adr, size_t len, uint16_t reg)
+{
+    traceRecs.emplace_back(TraceRec_t(MemhSendLoad, adr, len, reg));
+}
+
+void RevTracer::memReadResponse(size_t len, void *data, const MemReq* req)
+{
+    if (req->DestReg==0) return;
+    CompletionRec_t c;
+    c.hart = req->Hart;
+    c.destReg = req->DestReg;
+    c.len = len;
+    c.addr = req->Addr;
+    c.data = *((uint64_t*) data); // First 8 bytes only
```

Comment:
This may cause a bus error for unaligned data, and violate strict aliasing rules. Use:
```c++
memcpy(&c.data, data, sizeof(c.data));
```
(The `sizeof` is used so that if `c.data` changes size, the code adapts automatically.)